### PR TITLE
Improved File Viewer

### LINKED ISSUE #180: Prettier CSV/TSV File Preview
Context:
Current TSV/CSV files are viewed as raw txt. Would be nice to view this in a proper table format (such as [GitHub TSV render](https://github.com/umccr/NGS_Utils/blob/master/ngs_utils/reference_data/key_genes/umccr_cancer_genes.latest.tsv)). The initial [implementation](https://github.com/umccr/data-portal-client/blob/adb44b20620cac2a1f607259ddad673a422b39c2/src/components/PreviewActionButton.tsx#L321-L383) was done but removed due to the edge cases:
- Some tsv/csv is not on correct format. Ideally all rows from have the same number of commas or tabs, but some files in the portal had a missing comma/tabs in the middle of the row
- Files such as `MetricOutput.tsv` or `CombinedVariant.tsv` does have a header content defined as a square bracket in middle of the row.

TODO:
Implement / find a library that could handle the edge cases.

### PR DESCRIPTION
**Pull request by williamputraintan**
- Upgrade on TSV/CSV file previewer

Screenshot 2023-02-14 at 13 03 04
Screenshot 2023-02-14 at 13 03 04

Data taken from [UMCCR/NGS_Utils](https://github.com/umccr/NGS_Utils/blob/master/ngs_utils/reference_data/key_genes/umccr_cancer_genes.latest.tsv)

Fix #180

### CODE REVIEW

**Review by victorskl — APPROVED** [review]
Minor comment to attend. LGTM otherwise!

_(Oi! We are pretty close 🤏 to Excel/Spreadsheet view soon!)_

File: src/components/ViewPresignedUrl/index.tsx [hunk 1]
```diff
@@ -48,106 +51,154 @@ function ViewPresignedUrl({ presingedUrl }: Props) {
...
+      const rowData = rowString.split(delimiter);
+      return <div>{rowData[parseInt(prop.field)]} </div>;
+    };
+    const rowNumTemplate = (_: string, prop: { rowIndex: number }) => {
+      return <pre className='m-0'>{prop.rowIndex + 2} </pre>;
+    };
+    const [isTablePreview, setIsTablePreview] = useState<boolean>(true);
+    const options = ['Table', 'Raw'];
+    return (
+      <div className='w-full m-3 text-center' style={{ maxHeight: '80vh' }}>
```
[the remark this answers is not in the document]
Comment (williamputraintan) [reply src/components/ViewPresignedUrl/index.tsx:169]: Yup, changing it ...

**Comment by pdiakumis** [comment]
This looks awesome ;-)

**Comment by williamputraintan** [comment]
> This looks awesome ;-)

Thanks you!

**Pull request merged**